spec/models/note_spec.rb

```diff
@@ -1,6 +1,158 @@
+          expect { note.save! }.to raise_error(ActiveRecord::RecordInvalid)
+        end
+      end
+    end
+  end
+
+  describe '#word_count' do
+    let(:note) { build(:note, content: content) }
+
+    context 'when content is nil' do
+      let(:content) { nil }
+
+      it 'returns 0' do
+        expect(note.word_count).to eq(0)
+      end
+    end
+
+    context 'when content contains one word' do
+      let(:content) { 'Hello' }
+
+      it 'returns 1' do
+        expect(note.word_count).to eq(1)
+      end
+    end
+
+    context 'when content contains multiple words' do
+      let(:content) { 'Hello world! im doing test in ruby.' }
+
+      it 'returns the correct word count' do
+        expect(note.word_count).to eq(7)
+      end
+    end
+
+    context 'when content contains multiple spaces' do
+      let(:content) { 'Hello    world' }
+
+      it 'counts words correctly, ignoring extra spaces' do
+        expect(note.word_count).to eq(2)
+      end
+    end
+  end
+
+  describe '#content_length' do
+    context 'when user has NorthUtility' do
+      let(:utility) { create(:north_utility) }
+
+      context 'with short content' do
+        let(:content) { rand(0..utility.short_content) }
+
+        it 'returns "short" if word count is equal to or less than short_content' do
+          expect(note.content_length).to eq('short')
+        end
+      end
+
+      context 'with medium content' do
+        let(:content) { rand(utility.short_content + 1..utility.medium_content) }
+
+        it 'returns "medium" if word count is greater than short_content and equal to or less than medium_content' do
+          expect(note.content_length).to eq('medium')
+        end
+      end
+
+      context 'with long content' do
+        let(:content) { rand(utility.medium_content + 1..999) }
+
+        it 'returns "long" if word count is greater than medium_content' do
+          expect(note.content_length).to eq('long')
+        end
+      end
+    end
+
+    context 'when user has SouthUtility' do
+      let(:utility) { create(:south_utility) }
+
+      context 'with short content' do
+        let(:content) { rand(0..utility.short_content) }
+
+        it 'returns "short" if word count is equal to or less than short_content' do
+          expect(note.content_length).to eq('short')
+        end
+      end
+
+      context 'with medium content' do
+        let(:content) { rand(utility.short_content + 1..utility.medium_content) }
+
+        it 'returns "medium" if word count is greater than short_content and equal to or less than medium_content' do
+          expect(note.content_length).to eq('medium')
+        end
+      end
+
+      context 'with long content' do
+        let(:content) { rand(utility.medium_content + 1..999) }
+
+        it 'returns "long" if word count is greater than medium_content' do
+          expect(note.content_length).to eq('long')
+        end
+      end
+    end
+  end
```

Comment:
aplica el mismo comentario para todos estos `it`

Reply:
Perfecto ya cambie los it